internal/mempool/mempool.go

```diff
@@ -287,17 +287,17 @@ func (txmp *TxMempool) CheckTx(
 	}
 
 	wtx := &WrappedTx{
-		tx:        tx,
-		hash:      txHash,
-		timestamp: time.Now().UTC(),
-		height:    txmp.height,
+		tx:              tx,
+		hash:            txHash,
+		timestamp:       time.Now().UTC(),
+		height:          txmp.height,
+		expiredCallback: res.ExpireTxHandler,
```

Comment:
we might be able to simplify all the callback and passing-func-as-arg logics below by just having:
```
expiredCallback: func(removeFromCache bool) {
    if removeFromCache {
        txmp.cache.Remove(tx)
    }
    if res.ExpireTxHandler != nil {
        res.ExpireTxHandler()
    }
}
```
then we don't need to pass a function to `PurgeExpired` and can just call `ptx.expiredCallback(removeFromCache)` inside `PurgeExpired` (would need to pass removeFromCache though but it's still easier to understand i think

Reply:
:+1: I like that better